cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -884,6 +885,101 @@ public void deleteFlowQueues(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @POST
+  @Path("/apps/{app-id}/workflows/{workflow-name}/{run-id}/suspend")
+  public void suspendWorkflowRun(HttpRequest request, final HttpResponder responder,
+                                 @PathParam("namespace-id") String namespaceId, @PathParam("app-id") String appId,
+                                 @PathParam("workflow-name") String workflowName, @PathParam("run-id") String runId) {
+    try {
+      Id.Program id = Id.Program.from(namespaceId, appId, ProgramType.WORKFLOW, workflowName);
+      ProgramRuntimeService.RuntimeInfo runtimeInfo = findAllRuntimeInfos(namespaceId, appId,
+                                                                          ProgramType.WORKFLOW, workflowName,
+                                                                          runtimeService).get(RunIds.fromString(runId));
+      if (runtimeInfo == null) {
+        sendInvalidResponse(responder, id);
+        return;
+      }
+      AppFabricServiceStatus status = suspendResumeProgram(runtimeInfo.getController(), "suspend");
+      responder.sendString(status.getCode(), status.getMessage());
+    }  catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (Throwable e) {
+      LOG.error("Got exception:", e);
+      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @POST
+  @Path("/apps/{app-id}/workflows/{workflow-name}/{run-id}/resume")
+  public void resumeWorkflowRun(HttpRequest request, final HttpResponder responder,
+                                 @PathParam("namespace-id") String namespaceId, @PathParam("app-id") String appId,
+                                 @PathParam("workflow-name") String workflowName, @PathParam("run-id") String runId) {
+
+    try {
+      Id.Program id = Id.Program.from(namespaceId, appId, ProgramType.WORKFLOW, workflowName);
+      ProgramRuntimeService.RuntimeInfo runtimeInfo = findAllRuntimeInfos(namespaceId, appId,
+                                                                          ProgramType.WORKFLOW, workflowName,
+                                                                          runtimeService).get(RunIds.fromString(runId));
+      if (runtimeInfo == null) {
+        sendInvalidResponse(responder, id);
+        return;
+      }
+      AppFabricServiceStatus status = suspendResumeProgram(runtimeInfo.getController(), "resume");
+      responder.sendString(status.getCode(), status.getMessage());
+    }  catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (Throwable e) {
+      LOG.error("Got exception:", e);
+      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private void sendInvalidResponse(HttpResponder responder, Id.Program id) {
+    try {
+      AppFabricServiceStatus status;
+      try {
```

Comment:
This inner try-catch is not needed since you already have a catch block on the outer try that catches a ``Throwable`` and sends an ``INTERNAL_SERVER_ERROR``?